src/outputgeneration/ParseTreeMapWriter.js

```diff
@@ -114,7 +114,8 @@ export class ParseTreeMapWriter extends ParseTreeWriter {
       line: line,
       column: position.column || 0  // source map uses zero based columns
     };
-    if (position.source.name !== this.sourceName_) {
+    if (position.source.name !== this.rawSourceName_) {
```

Comment:
Please rename `this.sourceName_` to `this.relativeSourceName_` and use `this.sourceName_` for your fix.